engine/apps/mobile_app/views.py

```diff
@@ -72,3 +85,119 @@ def notification_timing_options(self, request):
             {"value": item[0], "display_name": item[1]} for item in MobileAppUserSettings.NOTIFICATION_TIMING_CHOICES
         ]
         return Response(choices)
+
+
+class MobileAppGatewayView(APIView):
+    authentication_classes = (MobileAppAuthTokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    class SupportedDownstreamBackends:
+        INCIDENT = "incident"
+
+    ALL_SUPPORTED_DOWNSTREAM_BACKENDS = [
+        SupportedDownstreamBackends.INCIDENT,
+    ]
+
+    def _determine_grafana_incident_api_url(self, organization: "Organization") -> typing.Optional[str]:
+        """
+        If the organization already has the Grafana Incident backend URL saved, use that.
+        Otherwise, ask the Grafana API for the URL from the Incident plugin's settings, then persist it.
+        """
+        if organization.grafana_incident_backend_url:
+            return organization.grafana_incident_backend_url
```

Comment:
this makes the assumption that a stack's Incident backend URL never changes. The alternative would be to move this logic to our sync and update things there instead